packages/geolocator/tizen/src/location_manager.cc

```diff
@@ -0,0 +1,164 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "location_manager.h"
+
+#include "log.h"
+
+LocationManager::LocationManager() {
+  location_manager_create(LOCATIONS_METHOD_HYBRID, &manager_);
+  location_manager_create(LOCATIONS_METHOD_HYBRID,
+                          &manager_for_current_location_);
+}
+
+LocationManager::~LocationManager() {
+  location_manager_destroy(manager_);
+  location_manager_destroy(manager_for_current_location_);
+}
+
+bool LocationManager::IsLocationServiceEnabled() {
+  bool gps_enabled = false;
+  int ret =
+      location_manager_is_enabled_method(LOCATIONS_METHOD_GPS, &gps_enabled);
+  if (LOCATIONS_ERROR_NONE != ret) {
+    throw LocationManagerError(ret);
+  }
+
+  bool wps_enabled = false;
+  ret = location_manager_is_enabled_method(LOCATIONS_METHOD_WPS, &wps_enabled);
+  if (LOCATIONS_ERROR_NONE != ret) {
+    throw LocationManagerError(ret);
+  }
+  return gps_enabled || wps_enabled;
+}
+
+Position LocationManager::GetLastKnownPosition() {
+  double altitude = 0.0;
+  double latitude = 0.0;
+  double longitude = 0.0;
+  double climb = 0.0;
+  double direction = 0.0;
+  double speed = 0.0;
+  double horizontal = 0.0;
+  double vertical = 0.0;
+  location_accuracy_level_e level = LOCATIONS_ACCURACY_NONE;
+  time_t timestamp = 0;
+
+  int ret = location_manager_get_last_location(
+      manager_, &altitude, &latitude, &longitude, &climb, &direction, &speed,
+      &level, &horizontal, &vertical, &timestamp);
+  if (LOCATIONS_ERROR_NONE != ret) {
+    throw LocationManagerError(ret);
+  }
+  Position position;
+  position.longitude = longitude;
+  position.latitude = latitude;
+  position.timestamp = timestamp;
+  position.accuracy = horizontal;
+  position.altitude = altitude;
+  position.heading = direction;
+  position.speed = speed;
+
+  return position;
+}
+
+void LocationManager::GetCurrentPosition(
+    LocationCallback on_location_callback,
+    LocationErrorCallback on_error_callback) {
+  on_location_callback_ = on_location_callback;
```

Comment:
Callback arguments can be named like `on_foo` or `foo_callback` but you shouldn't mix the two.

```suggestion
    LocationCallback on_location_update,
    LocationErrorCallback on_error) {
  location_callback_ = on_location_update;
```

This is what I meant in https://github.com/flutter-tizen/plugins/pull/387#discussion_r880054471. I don't think we need to rename `LocationCallback` to `LocationUpdateCallback` and `location_callback_` to `location_update_callback_`.

You might also refer to https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#naming-rules-for-typedefs-and-function-variables although it's not specifically for C++ code.

Reply:
@swift-kim
Thank you for detail review.
I have a question. about "the method that is called" in link.
According to the link, the `location_callback_` you suggest should be `handle_location_`, shouldn't it?

And there are two cases of using LocationCallback. (stream, getCurrentPosition).
So I think this is appropriate. What do you think?
```cpp
LocationCallback handle_location_update_;
LocationCallback handle_location_;
...
StartListenLocationUpdate(LocationCallback on_location_update)
handle_location_update_ = on_location_update;
...
GetCurrentPosition(LocationCallback on_location,LocationErrorCallback on_error
handle_location_ = on_location;
```

Reply:
> `location_callback_` should be `handle_location_`, shouldn't it?

No, `handleFoo` in the document refers to an explicit method or function like this:

```c++
static void HandleFoo(void* value) {
  // Handle the value.
}

SetFooHandler(HandleFoo);
```

> What do you think?

Yes, that looks good but the data member names would be:

```cpp
LocationCallback location_callback_;
LocationCallback location_update_callback_;
```

Reply:
Thank you for guide. I changed it